File: src/main/java/pl/mg/tasks/manager/model/Element.java
package pl.mg.tasks.manager.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;

import javax.persistence.*;

import static javax.persistence.GenerationType.IDENTITY;

@Entity
@Table(name = "elements")
@EqualsAndHashCode
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class Element {

    @Id
    @GeneratedValue(strategy = IDENTITY)
    private Integer id;

    @Column(name = "element_name")
    private String name;

    @Column(name = "element_details")
    private String details;

    @ManyToOne
    @JoinColumn(name = "task")
    @JsonIgnore
    private Task task;
}

File: frontend/src/app/model/element.ts
export class Element {

  public _id: number;
  public _name: string;
  public _details: string;

  constructor(id: number, name: string, details: string) {
    this._id = id;
    this._name = name;
    this._details = details;
  }

  get id(): number {
    return this._id;
  }

  get name(): string {
    return this._name;
  }

  get details(): string {
    return this._details;
  }

}
